pkgs/tools/package-management/cargo-edit/default.nix

```
@@ -4,25 +4,27 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "cargo-edit";
-  version = "0.5.0";
+  version = "0.6.0";
 
   src = fetchFromGitHub {
     owner = "killercup";
     repo = pname;
     rev = "v${version}";
-    sha256 = "16gpljbzk6cibry9ssnl22xbcsx2cr57mrs3x3n6cfmldbp6bhbr";
+    sha256 = "19jnvsbddn52ibjv48jyfss25gg9mmvxzfhbr7s7bqyf3bq68jbm";
   };
 
-  cargoSha256 = "1zwkar914zyghky09lgk0s374m5d6yccn0m15bqlgxxyymg4b59y";
+  cargoSha256 = "0b06jsilj87rnr1qlarn29hnz0i9p455fdxg6nf6r2fli2xpv1f0";
 
   nativeBuildInputs = [ pkg-config ];
   buildInputs = [ openssl ] ++ lib.optionals stdenv.isDarwin [ libiconv darwin.apple_sdk.frameworks.Security ];
 
+  checkPhase = false; # integration tests depend on changing cargo config
```

It's also setting `doCheck` instead of clobbering the `checkPhase` implementation.

```suggestion
  doCheck = false; # integration tests depend on changing cargo config
```

---

Updated.

Is this documented somewhere, btw? I only copied this from some other package (sccache, I think).

---

In the manual section describing all the phases, it mentions the `doCheck` attribute:
https://nixos.org/nixpkgs/manual/#ssec-check-phase

In general, most stdenv phases have `doFoo` to skip them. This leaves the implementation there for inspection in the derivation, and allows an overlay to swap the boolean back on to run the implementation if they're fixing/patching it.  

The Rust helper defaults it to true, even though the lower level stdenv doesn't. This should probably be documented in the Rust manual section, if you're interested (in another PR, of course, since it's not blocking this change!):
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/rust.section.md

The python language helpers behave similarly to the Rust helpers, and have a nice manual section describing their stuff, for reference:
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-function

The markdown source file is compiled into the manual you see on https://nixos.org/nixpkgs/manual/#rust

---

Very interesting -- I didn't get the relation how the different package builders use these phases. I'll read some more on that and see if I find time to update the manual :)